lib/graphql/schema/validation.rb

```diff
@@ -100,7 +100,8 @@ def self.assert_named_items_are_valid(item_name, get_items_proc)
 
         DEFAULT_VALUE_IS_VALID_FOR_TYPE = -> (type) {
           if !type.default_value.nil?
-            coerced_value = type.type.coerce_input(type.default_value)
+            coerced_value = type.type.coerce_result(type.default_value)
```

Comment:
There were a number of tests that would setup arguments like:
```ruby
argument :invalid do
  type GraphQL::STRING_TYPE
  default_value [1,2,3]
end
```
And then assert:
```ruby
it "applies validation to its member Arguments" do
  assert_error_includes invalid_argument_member_field, "default value [1, 2, 3] is not valid for type String"
end
```

This behavior is nice, because it's an easy way to catch developer mistakes. Also, it's still possible to preserve it, because input values are always eventually composed of scalars. Just need to add `coerce_result` behavior for lists and input object types.